File: apps/web/types/approval.ts
/** Approval types matching the backend ApprovalRead schema. */

export type ApprovalStatus = "pending" | "approved" | "rejected";

export interface Approval {
  id: string;
  status: ApprovalStatus;
  title: string;
  description: string | null;
  project_id: string;
  run_id: string | null;
  task_id: string | null;
  artifact_id: string | null;
  decided_by: string | null;
  decision_comment: string | null;
  decided_at: string | null;
  created_at: string;
  updated_at: string;
}

export interface ApprovalList {
  items: Approval[];
  total: number;
}

File: apps/api/app/schemas/approval.py
"""Approval schemas — request and response models for approval workflow."""

import uuid
from datetime import datetime

from pydantic import BaseModel, Field

from app.models.approval_request import ApprovalStatus


# ---------- Response schemas ----------
class ApprovalRead(BaseModel):
    id: uuid.UUID
    status: ApprovalStatus
    title: str
    description: str | None
    project_id: uuid.UUID
    run_id: uuid.UUID | None
    task_id: uuid.UUID | None
    artifact_id: uuid.UUID | None
    decided_by: str | None
    decision_comment: str | None
    decided_at: datetime | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ApprovalList(BaseModel):
    items: list[ApprovalRead]
    total: int


# ---------- Request schemas ----------
class ApprovalCreate(BaseModel):
    title: str = Field(..., min_length=1, max_length=500)
    description: str | None = None
    project_id: uuid.UUID
    run_id: uuid.UUID | None = None
    task_id: uuid.UUID | None = None
    artifact_id: uuid.UUID | None = None


class ApprovalDecision(BaseModel):
    status: ApprovalStatus
    decided_by: str | None = Field(None, max_length=100)
    decision_comment: str | None = None
